### ISSUE TITLE
Only open single vbucket stream for cfg document

### ISSUE THREAD

**Issue by tleyden** [couchbaselabs/cbgt#10]
Basically this was from a conversation between @steveyen and @adamcfraser where Adam pointed out that since CBGT is only storing a single config document `cfg`, and then opening up a DCP stream to listen for changes to that config document, it's possible to optimize it such that you just figure out the vbucket id of the config document and open up a DCP stream to _only_ that vbucket, in order to lower the overall DCP overhead.

**Comment by mschoch** [comment]
i think i might need a description to understand this one

**Comment by tleyden** [comment]
Sorry, was on the phone and just stubbed this out so I didn't forget.  See above.

**Comment by tleyden** [comment]
@steveyen this is a bit of a blocker for us and shouldn't be a lot of work.  Can you take a look?

**Comment by steveyen** [comment]
Took a quick look, and notes/ideas to implement this: probably just need to call Bucket.VBHash(cfgKey) during NewCfgCB()...

```
https://godoc.org/github.com/couchbase/go-couchbase#Bucket.VBHash
```

Also, briefly talked with @tleyden about priority.  It's more of an expected future performance ship blocker, but not a "blocker" right now as in stopping any current work of anybody.  So, got some headroom of couple weeks to turnaround instead of hours.

One thing pops to mind, probably also need to implement this optimization about the same time as the breaking API change of https://github.com/couchbaselabs/cbgt/issues/19

Related, from looking at the NewCfgCB() code, I see there's a race around where it initializes the cbdatasource, instead of waiting for the SetKeyPrefix().

**Comment by steveyen** [comment]
fixed in commit 8fec866
